backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/pageindicator/BpkPageIndicator.kt

```diff
@@ -0,0 +1,200 @@
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.progressSemantics
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.platform.LocalLayoutDirection
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.semantics.setProgress
+import androidx.compose.ui.semantics.stateDescription
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.LayoutDirection
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import java.lang.Integer.min
+import kotlin.math.roundToInt
+
+enum class BpkPageIndicatorStyle {
+  Default,
+  OverImage,
+}
+
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+fun BpkPageIndicator(
+  currentIndex: Int,
+  totalIndicators: Int,
+  indicatorLabel: String,
+  modifier: Modifier = Modifier,
+  style: BpkPageIndicatorStyle = BpkPageIndicatorStyle.Default,
+) {
+
+  if (totalIndicators <= 1) return
+  val indicatorSize = BpkSpacing.Md
+  val indicatorCount = min(totalIndicators, DISPLAY_DOTS_MAX)
+  val transitionDistance = LocalDensity.current.run { (indicatorSize * 2).toPx().toInt() }
+  val rtlMultiplier = if (LocalLayoutDirection.current == LayoutDirection.Rtl) -1 else 1
+
+  AnimatedContent(
+    modifier = modifier.pageIndicatorModifier(
+      indicatorCount = indicatorCount,
+      indicatorSize = indicatorSize,
+      range = 0 until totalIndicators,
+      currentIndex = currentIndex,
+      indicatorLabel = indicatorLabel,
+    ),
+    targetState = currentIndex,
+    transitionSpec = {
+      val isIncreasing = targetState >= initialState
+      val multiplier = if (isIncreasing) 1 else -1
+      val indexRange = if (isIncreasing) 3..totalIndicators - 3 else 2..totalIndicators - 4
+      val shouldAnimate = totalIndicators > DISPLAY_DOTS_MAX && currentIndex in indexRange
+      ContentTransform(
+        targetContentEnter = if (shouldAnimate) slideInHorizontally(
+          animationSpec = tween(TRANSITION_DURATION),
+          initialOffsetX = { transitionDistance * multiplier * rtlMultiplier },
+        ) else EnterTransition.None,
+        initialContentExit = if (shouldAnimate) slideOutHorizontally(
+          animationSpec = tween(TRANSITION_DURATION),
+          targetOffsetX = { transitionDistance * -multiplier * rtlMultiplier },
+        ) else ExitTransition.None,
+      )
+    },
+  ) { index ->
+    PageIndicatorRow(
+      indicatorSize = indicatorSize,
+      totalIndicators = totalIndicators,
+      index = index,
+      style = style,
+    )
+  }
+}
+
+@Composable
+private fun PageIndicatorRow(
+  indicatorSize: Dp,
+  totalIndicators: Int,
+  index: Int,
+  style: BpkPageIndicatorStyle,
+  modifier: Modifier = Modifier,
+) {
+  Row(
```

Comment:
I decided to use a Row instead of LazyRow because scrolling and selecting an item on it doesn't work correctly. i.e. it always selects item 0

Reply:
The current approach does a lot of custom stuff, especially with animations. Maintaining this might be more problematic that working with LazyRow. Can you show a sample of the code with LazyRow which doesn't work?

Reply:
https://github.com/Skyscanner/backpack-android/tree/sonic/SONIC-1204-BpkPageIndicator-LazyRow
This branch is implemented using LazyRow but it's not working as expected

Reply:
PageIndicator in web is also using a list and animates it according to selected position
https://github.com/Skyscanner/backpack/blob/main/packages/bpk-component-page-indicator/src/BpkPageIndicator.module.scss#L49

Reply:
done